lib/carbon/cache.py

```diff
@@ -236,6 +237,10 @@ def get_datapoints(self, metric):
     """Return a list of currently cached datapoints sorted by timestamp"""
     return sorted(self.get(metric, {}).items(), key=by_timestamp)
 
+  def pop_new_metric(self):
+    # return first seen metric
+    return self.new_metrics.pop(0)
```

Comment:
this can raise "IndexError: pop from empty list"
(which does not appear to be handled where this is called)

Reply:
Thanks, it was caught later on and not logged so it got lost. Changed the code so we log exceptions and handled this.